### PR TITLE
Added Timestamp to EventEnvelope

### LINKED ISSUE #4668: Add Timestamp and missing properties to Akka.Persistence.Query.EventEnvelope
This information is sometimes stored or derived as part of the `Offset` on the `EventEnevelope`: 

https://github.com/akkadotnet/akka.net/blob/77babf022eb7c06ecd8d12919af2441fed3b2fe1/src/core/Akka.Persistence.Query/Offset.cs#L50-L84

However, it looks to me like this is an oversight in the design of the `EventEnvelope`. There's no reason it shouldn't have an explicit timestamp appended to it given that this data is commonplace inside the Akka.Persistence journal.

We should create an issue and add some constructor overloads (can't modify the existing signatures - breaks binary compatibility) that time a Timestamp and exposed that as a public property on `EventEnvelope`.

_Originally posted by @Aaronontheweb in https://github.com/akkadotnet/akka.net/discussions/4665#discussioncomment-205528_

### PR DESCRIPTION
**Pull request by ismaelhamed**
Closes #4668

### CODE REVIEW

File: src/core/Akka.Persistence.TCK/Akka.Persistence.TCK.csproj [hunk 1]
```diff
@@ -16,6 +16,10 @@
     <ProjectReference Include="..\Akka.Streams.TestKit\Akka.Streams.TestKit.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Include="FluentAssertions" Version="$(FluentAssertionsVersion)" />
```
Comment (Aaronontheweb) [inline src/core/Akka.Persistence.TCK/Akka.Persistence.TCK.csproj:20]: we should probably just add a dependency to this in the TestKit itself - but that's an issue for another day

File: src/core/Akka.Persistence.TCK/Query/CurrentEventsByPersistenceIdSpec.cs [hunk 2]
```diff
@@ -165,6 +166,23 @@ public virtual void ReadJournal_CurrentEventsByPersistenceId_should_return_empty
             var src = queries.CurrentEventsByPersistenceId("l", 4L, 3L);
             src.Select(x => x.Event).RunWith(this.SinkProbe<object>(), Materializer).Request(1).ExpectComplete();
         }
+        
+                
+        [Fact]
+        public void ReadJournal_CurrentEventsByPersistenceId_should_include_timestamp_in_EventEnvelope()
```
Comment (Aaronontheweb) [inline src/core/Akka.Persistence.TCK/Query/CurrentEventsByPersistenceIdSpec.cs:172]: LGTM

File: src/protobuf/Persistence.proto [hunk 3]
```diff
@@ -14,6 +14,7 @@ message PersistentMessage {
   string sender = 11; // not stored in journal, needed for remote serialization 
   string manifest = 12;
   string writerGuid = 13;
+  sint64 timestamp = 14;
```
Comment (Aaronontheweb) [inline src/protobuf/Persistence.proto:17]: LGTM

**Review by Aaronontheweb — APPROVED** [review]
LGTM - but I need to check on whether or not that change to `Persistent` is going to be an issue... Shouldn't take long

File: src/core/Akka.API.Tests/CoreAPISpec.ApprovePersistence.approved.txt [hunk 4]
```diff
@@ -413,13 +415,14 @@ namespace Akka.Persistence
     [Akka.Annotations.InternalApiAttribute()]
     public class Persistent : Akka.Persistence.IPersistentRepresentation, Akka.Persistence.Serialization.IMessage, System.IEquatable<Akka.Persistence.IPersistentRepresentation>
     {
-        public Persistent(object payload, long sequenceNr = 0, string persistenceId = null, string manifest = null, bool isDeleted = False, Akka.Actor.IActorRef sender = null, string writerGuid = null) { }
+        public Persistent(object payload, long sequenceNr = 0, string persistenceId = null, string manifest = null, bool isDeleted = False, Akka.Actor.IActorRef sender = null, string writerGuid = null, long timestamp = 0) { }
```
[the remark this answers is not in the document]
Comment (Aaronontheweb) [reply src/core/Akka.API.Tests/CoreAPISpec.ApprovePersistence.approved.txt:418]: Yeah, looks like this is going to be an issue https://github.com/akkadotnet/Akka.Persistence.MongoDB/blob/4bd8ca146c7433aa875681062260b9a05fbc4dff/src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs#L354-L409

We should patch this. I'll merge the PR as-is and update it myself.

[the remark this answers is not in the document]
Comment (CumpsD) [reply src/core/Akka.API.Tests/CoreAPISpec.ApprovePersistence.approved.txt:418]: @Aaronontheweb does it impact https://github.com/AkkaNetContrib/Akka.Persistence.PostgreSql too?

**Comment by Aaronontheweb** [comment]
Yes it will - need to recompile

**Comment by CumpsD** [comment]
Thanks, that's an easy fix :)

File: src/core/Akka.API.Tests/CoreAPISpec.ApprovePersistence.approved.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (ia64mail) [reply src/core/Akka.API.Tests/CoreAPISpec.ApprovePersistence.approved.txt:418]: @Aaronontheweb Any updates on the roll out plans for a new version of Akka.Persistence.MongoDB? We have Akka core v.1.4.14 and MongoDB v.1.4.12 and stuck with the error during application startup persistent records recovery: 

```
Method not found: 'Void Akka.Persistence.Persistent..ctor(System.Object, Int64, System.String, System.String, Boolean, Akka.Actor.IActorRef, System.String)'.
   at Akka.Persistence.MongoDb.Journal.MongoDbJournal.ToPersistenceRepresentation(JournalEntry entry, IActorRef sender)
   at Akka.Persistence.MongoDb.Journal.MongoDbJournal.<>c__DisplayClass10_0.<ReplayMessagesAsync>b__4(JournalEntry doc)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Akka.Persistence.MongoDb.Journal.MongoDbJournal.<ReplayMessagesAsync>d__10.MoveNext()
```

Would appreciate any details regarding the timeline

**Pull request merged**